### PR TITLE
feat(dashboard): order edits in timeline

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- add order edit confirmed/created events in the timeline
- add order change endpoint clients
- panel for active edit and pending edit
- few fixes around the edit domain

### CODE REVIEW

File: packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-activity-section/order-timeline.tsx [hunk 1]
```diff
@@ -686,3 +712,68 @@ const ExchangeBody = ({
...
+
+function countItemsChange(
+  actions: AdminOrderChange["actions"],
+  itemsMap: Record<string, AdminOrderLineItem>
+) {
+  let added = 0
+  let removed = 0
+
+  actions.forEach((action) => {
+    if (action.action === "ITEM_ADD") {
```
Comment (carlos-r-l-rodrigues) [inline packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-activity-section/order-timeline.tsx:763]: I think we could have a const/enum with all the actions we check everywhere, if there are other checks.

File: packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-activity-section/order-timeline.tsx [hunk 2]
```diff
@@ -304,6 +311,25 @@ const useActivityItems = (order: AdminOrder): Activity[] => {
...
 
+    for (const edit of orderChanges) {
+      const isConfirmed = edit.status === "confirmed"
+
+      items.push({
+        title: t(`orders.activity.events.edit.${edit.status}`, {
+          editId: edit.id.slice(-7),
+        }),
+        timestamp:
+          edit.status === "crated"
```
Comment (carlos-r-l-rodrigues) [inline packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-activity-section/order-timeline.tsx]:
```suggestion
          edit.status === "created"
```

File: packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-activity-section/order-timeline.tsx [hunk 3]
```diff
@@ -304,6 +311,25 @@ const useActivityItems = (order: AdminOrder): Activity[] => {
...
+            ? edit.created_at
+            : edit.status === "requested"
+            ? edit.requested_at
+            : edit.canceled_at,
```
Comment (carlos-r-l-rodrigues) [inline packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-activity-section/order-timeline.tsx]: we can also handle the `declined` status as well

File: packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-activity-section/order-timeline.tsx (hunk 1 above)
Comment (fPolic) [reply packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-activity-section/order-timeline.tsx:763]: yup, we have these constants hardcoded in a few places, will refactor this in a separate PR

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Pull request merged**